src/main/java/com/truelayer/java/http/mappers/ErrorMapper.java

```diff
@@ -30,16 +31,21 @@
             return buildFallbackError(response.code(), correlationId, GENERIC_ERROR_TITLE);
         }
 
+        ProblemDetails problemDetails;
         try {
-            return Utils.getObjectMapper().readValue(errorBodyString, ProblemDetails.class);
+            problemDetails = getObjectMapper().readValue(errorBodyString, ProblemDetails.class);
         } catch (JsonProcessingException e) {
             return tryMapLegacyError(response.code(), correlationId, errorBodyString);
         }
+
+        return problemDetails.isWellFormed()
```

Comment:
this is required as a response completely different from a Problem details would not leady to a desieralization exception